native/cocos/renderer/GFXDeviceManager.h

```diff
@@ -142,15 +142,17 @@ class CC_DLL DeviceManager final {
         Device *device = CC_NEW(DeviceCtor);
 
         if (DETACH_DEVICE_THREAD) {
+            CC_DELETE(device);
             device = CC_NEW(gfx::DeviceAgent(device));
         }
 
         if (CC_DEBUG > 0 && !FORCE_DISABLE_VALIDATION || FORCE_ENABLE_VALIDATION) {
+            CC_DELETE(device);
             device = CC_NEW(gfx::DeviceValidator(device));
```

Comment:
After CC_DELETE device , device is an invalid pointer

Reply:
Yep, can not delete it.